include/parser.h

```diff
@@ -0,0 +1,36 @@
+#ifndef INCLUDE_PARSER_H_
+#define INCLUDE_PARSER_H_
+
+#include <stdio.h>
+#include <stdlib.h>
+
+/* Parsing Functions for Parsing Input Files */
+
+/*
+ * parse_file
+ * parameter: filename
+ * returns with pointer to array of strings, each of which represent a line in a given file
+ * current limit to each string is 4096 characters, excluding the newline
```

Comment:
This is because fgets has a buffer limit. I've been searching for ways to remove this limit, and am considering strtok and getchar. I'll remove this limit and update here.